docs/architecture/adr-002-lite-client.md

```diff
@@ -0,0 +1,454 @@
+the same height). 
+
+In what follows we outline the following components of the lite client software:
+
+- core lite client verification library and traits
+- how Tendermint data-structures implement the core lite client traits
+- how a lite node requests information from full nodes and detects conflicts
+
+Note that the architecture of IBC is out of scope, suffice it to say that the
+core libraries (ie. the verification library and the implementation of core
+traits by the Tendermint types) will be re-usable by IBC.
+
+## Decision
+
+Below is a schematic of the components of a lite node:
+
+![Lite Client Diagram](assets/adr-002-image.png)
+
+We take up the components of the diagram in term.
+
+### State
+
+The lite node state contains the following:
+
+- current height (H)
+- last header (H-1)
+- current validators (H) - including all validator pubkeys and voting powers
+
+It also includes some configuration, which contains:
+
+- trusting period
+- initial list of full nodes
+- method (sequential or skipping)
+- trust level (if method==skipping)
+
+The node is initialized with a trusted header for some height H-1
+(call this header[H-1]), and a validator set for height H (call this vals[H]).
+
+The node may be initialized by the user with only a height and header hash, and
+proceed to request the full header and validator set from a full node. This
+reduces the initialization burden on the user, and simplifies passing this
+information into the process, but for the state to be properly initialized it
+will need to get the correct header and validator set before starting the lite
+client syncing protocol.
+
+The configuration contains an initial list of peers.
+For the sake of simplicity, one of the peers is selected as the "primary", while the
+rest are considered "backups". Most of the data is downloaded from the primary,
+and double checked against the backups.
+
+The state is considered "expired" if the difference between the current time and
+the time from the trusted header is greater than a configurable "trusting
+period". If at any point the state is expired, the node should log an error and
+exit - it's needs to be manually reset.
+
+
+### Manager
+
+For lack of a better name. The Manager co-ordiantes the syncing and is the
+highest level component. We consider two approaches to syncing the lite node: sequential and skipping.
+
+#### Sequential Sync
+
+Inital state: 
+
+    - time T
+    - height H 
+    - header[H-1]
+    - vals[H]
+
+Here we describe the happy path:
+
+1) Request header[H], commit[H], and vals[H+1] from the primary, and check that they are well formed and from the correct height
+2) Pass header[H], commit[H], vals[H], and vals[H+1] to the verification library, which
+
+  will:
+    - check that vals[H] and vals[H+1] are correctly reflected in the header
+    - check that the commit is for the header
+    - check that +2/3 of the validators correctly signed the header hash
```

Comment:
A minor point I hadn't thought of before, not an actual suggestion for an immediate change:

The lite client could instead check that `1/3 + c` signed the header hash, and you get a sliding scale of guarantees in exchange for better efficiency. 

* When `c > 0`, the lite client is guaranteed that upon seeing conflicting satisfying commits, the `1/3` assumption has been broken. Upon learning this, the client can feel free to panic / abandon the chain until manual override, etc.
* For `epsilon < c < 1/3`, the lite client increases its odds of catching the double-signing node.
* When `c > 1/3`, the lite client is guaranteed that upon seeing conflicting satisfying commits, it can create a fraud proof for at least one entity.

Since seeing a successful fork is such an extreme event, it seems likely to me that many lite clients may find it sufficient to be aware of the fork and deal w/ it via social consensus, and to not have to be able to submit the fraud proof.

(This whole point is moot once aggregate signatures are in. Then theres no advantage to only verifying a partial amount)

Reply:
Cool - want to open an issue on the spec repo for this idea? We can also try to formally model the distinction in these guarantees: https://github.com/tendermint/spec

Reply:
Done! https://github.com/tendermint/spec/issues/59